src/bidiMapper/domains/context/logHandler.ts

```diff
@@ -0,0 +1,241 @@
+      throw new Error(
+        'More value is provided: "' +
+          this._getRemoteValuesText(args, false) +
+          '"'
+      );
+    }
+
+    return output;
+  }
+
+  /**
+   * @param arg input remote value to be parsed
+   * @returns parsed text of the remote value
+   *
+   * input: {"type": "number", "value": 1}
+   * output: 1
+   *
+   * input: {"type": "string", "value": "abc"}
+   * output: "abc"
+   *
+   * input: {"type": "object",  "value": [["id", {"type": "number", "value": 1}]]}
+   * output: '{"id": 1}'
+   *
+   * input: {"type": "object", "value": [["font-size", {"type": "string", "value": "20px"}]]}
+   * output: '{"font-size": "20px"}'
+   */
+  private _getSingleRemoteValueText(arg: CommonDataTypes.RemoteValue): string {
+    // arg type validation
+    if (!['number', 'string', 'object'].includes(arg.type)) {
+      throw Error('Invalid value type: ' + arg.toString());
+    }
+
+    if (arg.type === 'number' && typeof (arg.value, Number)) {
+      return arg.value.toString();
+    }
+
+    if (arg.type === 'string' && typeof (arg.value, String)) {
+      return '"' + arg.value.toString() + '"';
+    }
+
+    if (arg.type === 'object' && typeof (arg.value, Object)) {
+      return (
+        '{"' +
+        (arg.value as any[])[0][0] +
+        '": ' +
+        this._getSingleRemoteValueText((arg.value as any[])[0][1]) +
+        '}'
+      );
+    }
+
+    throw Error('Invalid value type: ' + arg.toString());
+  }
+
+  private _getRemoteValuesText(
+    args: CommonDataTypes.RemoteValue[],
+    formatText: boolean
+  ): string {
+    // if args[0] is a format specifier, format the args as output
+    if (
+      args[0].type === 'string' &&
+      this._isFormmatSpecifier(args[0].value.toString()) &&
+      formatText
+    ) {
+      return this._formatter(args);
+    }
+
+    // if args[0] is not a format specifier, just join the args with \u0020
+    return args
+      .map((arg) => (arg as { type: string; value: any }).value.toString())
+      .join('\u0020');
+  }
+
+  // convert CDP StackTrace object to Bidi StackTrace object
+  private _getBidiStackTrace(
+    cdpStackTrace: Protocol.Runtime.StackTrace | undefined
+  ): Script.StackTrace | undefined {
+    const stackFrames = cdpStackTrace?.callFrames.map((callFrame) => {
+      return {
+        columnNumber: callFrame.columnNumber,
+        functionName: callFrame.functionName,
+        lineNumber: callFrame.lineNumber,
+        url: callFrame.url,
+      };
+    });
+
+    const bidiStackTrace = stackFrames
+      ? {
+          callFrames: stackFrames,
+        }
+      : undefined;
+
+    return bidiStackTrace;
+  }
+
+  private _initializeLogEntryAddedEventListener() {
+    this._cdpClient.Runtime.on('consoleAPICalled', async (params) => {
+      const args = await Promise.all(
+        params.args.map(async (arg) => {
+          return await this._serializeCdpObject(arg);
+        })
+      );
+
+      await this._bidiServer.sendMessage({
```

Comment:
Shouldn't it be done via `eventManager`, which manages subscriptions?

Reply:
removed logHandler.ts.